Feat: added support for alternate css units for grid layout (#2659)
uber/baseweb: documentation-site/pages/components/layout-grid.mdx

If you want to customize the breakpoints of the grid, you can do so via the `breakpoints` and `mediaQuery` properties. The grid will always pull breakpoints from the theme and correlate responsive props with those breakpoints.

<Example title="Custom css units" path="layout-grid/custom-css-unit.js">
  <CustomCssUnitExample />
</Example>

> 

Author: I have added it back here.

If you want to customize the breakpoints of the grid, you can do so via the `breakpoints` and `mediaQuery` properties. The grid will always pull breakpoints from the theme and correlate responsive props with those breakpoints.

<Example title="Custom css units" path="layout-grid/custom-css-unit.js">
  <CustomCssUnitExample />
</Example>

> Layout grid uses `px` as a default css unit, you can change that by modifing theme in following manner:

Reviewer: ```suggestion
Layout grid uses `px` as its default CSS unit. You can change this by modifying the theme as in the example above.
```

```js
const theme = {
  grid: {
    unit: 'em',
  },
};

> ```

Reviewer: Can remove if we swap the example to show theme customizations.

Author: If I understood this correctly, this should be part of theme customisation section. right?

Author: I have made changes accordingly, Please let me know if I understood this wrong :/

> If you are using different css unit than pixle, you would also have to add `gutters` and `margins` in same css unit.For example if you are using `em` as a css unit `gutters` and `margins` might have value like below:

Reviewer: ```suggestion
Make sure to update any properties on the theme that use `unit`, namely `gutters`, `margins`, and `maxWidth`, as the defaults for these properties will likely not work well with any `unit` other than `px`.
```

```js
const theme = {
  grid: {
    unit: 'em',
    gutters: [1, 2, 2],
    margins: [1, 2, 4],
  },
};

> ```

Reviewer: Can remove once the example is updated.

Author: done 👍

> Grid component also accepts unit as a prop. By using this prop you can override theme's css unit for grid.

Reviewer: ```suggestion
You can also override the CSS unit for a grid on an instance-by-instance basis via the `unit` prop on `Grid`. 
```

> 

Reviewer: Maybe add a snippet here showing the instance level setting of unit on Grid.

```js
<Grid gridUnit="em" gridMargins={[1, 2, 3]} gridMaxWidth={40}>{/* ... */}</Grid>
```

Author: Done 👍

If you want to customize the breakpoints of the grid, you can do so via the `breakpoints` and `mediaQuery` properties. The grid will always pull breakpoints from the theme and correlate responsive props with those breakpoints.

<Example title="Custom css units" path="layout-grid/custom-css-unit.js">
  <CustomCssUnitExample />
</Example>

Layout grid uses `px` as a default css unit, If you are using different css unit than pixle, you would also have to add `gridMaxWidth`, `gutters` and `margins` in same css unit. For example if you are using `em` as a css unit `gridMaxWidth`, `gutters` and `margins` might have value like below:

```js
const theme = {
  grid: {
    unit: 'em',
    gutters: [1, 2, 2],
    margins: [1, 2, 4],
    gridMaxWidth: 70,
  },
};
```

Grid component also accepts unit as a prop. By using this prop you can override theme's CSS length unit for grid.

```
<Grid gridUnit="em" gridMargins={[1, 2, 3]} gridMaxWidth={40}>{/* ... */}</Grid>
```

Following are the CSS Length unit supported:

- `cm`
- `mm`
- `Q`
- `in`
- `pc`
- `px`
- `pt`
- `em`
- `ex`
- `ch`
- `rem`
- `lh`
- `vw`
- `vh`
- `vmin`
- `vmax`

> - `%`

Author: done 👍
